ui/src/ducks/app/nodes.js

```diff
@@ -246,21 +277,25 @@ export function* fetchNodes() {
             status = API_STATUS_UNKNOWN;
           }
 
+          // the Roles of the Node should be the ones that are stored in the labels `node-role.kubernetes.io/<role-name>`
+          let nodeRolesLabels = [];
           const roleTaintMatched = roleTaintMap.find((item) => {
-            const nodeRoles = Object.keys(node.metadata.labels).filter((role) =>
+            nodeRolesLabels = Object.keys(node.metadata.labels).filter((role) =>
               role.includes(ROLE_PREFIX),
             );
```

Comment:
Quoting my previous comment:

> Instead of filling `nodeRolesLabels` on each pass of `find` (which is highly inefficient), just define it when creating the `nodeRolesLabels` variable (which should be a `const` anyway).

That would look like:

```javascript
// the Roles of the Node should be the ones that are stored in the labels `node-role.kubernetes.io/<role-name>`
const nodeRolesLabels = Object.keys(node.metadata.labels).filter(
  (label) => label.startsWith(ROLE_PREFIX),
);
const roleTaintMatched = roleTaintMap.find((item) => {
  return (
    // what you already have
  );
};
```

Note that I used `startsWith` instead of `includes`, which is more accurate (`includes` could match a `custom.node-role.kubernetes.io/toto` label, which we don't want)

Reply:
Remove the `roleTaintMatched` and control_plane, workload_plane, bootstrap and infra from the returned Node object.
Since we can get the Roles from the Labels, I don't think we need those props anymore.